src/dotnet/APIView/APIViewWeb/Repositories/PullRequestManager.cs

```diff
@@ -151,59 +159,72 @@ private async Task<bool> IsReviewSame(ReviewModel review, RenderedCodeFile rende
+        {
+            var newRevision = new ReviewRevisionModel()
+            {
+                Author = pullRequestModel.Author,
+                Label = $"Created for PR {prNumber}"
+            };
             var stringBuilder = new StringBuilder();
             stringBuilder.Append(ISSUE_COMMENT_PACKAGE_IDENTIFIER.Replace("<PKG-NAME>", codeFile.PackageName));
             stringBuilder.Append(Environment.NewLine).Append(Environment.NewLine);
             // Get automatically generated master review for package or previously cloned review for this pull request
             var review = await GetBaseLineReview(codeFile.Language, codeFile.PackageName, pullRequestModel);
-            if (review == null)
-            {
-                return "";
-            }
-
-            // Check if API surface level matches with any revisions
-            var renderedCodeFile = new RenderedCodeFile(codeFile);
-            if (await IsReviewSame(review, renderedCodeFile))
+            if (review == null)
             {
-                //Do not update the comment if review was already created and it matches with current revision.
-                if (pullRequestModel.ReviewId != null)
-                    return "";
-
-                //Baseline review was not created earlier or this is the first commit of PR
-                stringBuilder.Append($"API changes are not detected in this pull request for `{codeFile.PackageName}`");
-                return stringBuilder.ToString();
-            }
-                
-
-            if(pullRequestModel.ReviewId != null)
-            {
-                // If baseline review was already created and if APIs in current commit doesn't match any of the revisions in generated review then create new baseline using main branch and compare again.
-                // If APIs are still different, find the diff against latest baseline.
-                review = await GetBaseLineReview(codeFile.Language, codeFile.PackageName, pullRequestModel, true);
-                review.ReviewId = pullRequestModel.ReviewId;
-                if (await IsReviewSame(review, renderedCodeFile))
+                // If base line is not available (possible if package is new or request coming from SDK automation)
+                review = new ReviewModel()
+                {
+                    Author = pullRequestModel.Author,
+                    CreationDate = DateTime.Now,
+                    Name = pullRequestModel.PackageName,
+                    IsClosed = false,
+                    FilterType = ReviewType.PullRequest,
+                    ReviewId = IdHelper.GenerateId()
+                };
+                var reviewUrl = reviewUrlPrefix + review.ReviewId;
+                stringBuilder.Append($"API review is created for `{codeFile.PackageName}`. You can review APIs [here]({reviewUrl}).").Append(Environment.NewLine);

```

Comment:
A quick question since I am not familiar with the code.

If a new PR on a new package, I will expect to see it gets to this "review == null" code path.

What if author of that PR then upload another commit (in same PR) that changed some code that affects API, does this gets to "review == null" code path again?

I think we will be fine, as long as the api review reflect latest change.

Reply:
It won't because we already have a review created and we track that as baseline.